tests/conftest.py

```diff
@@ -114,3 +114,52 @@ def json(self):
             return {'token': 'superlegittoken'}
 
     mocker.patch("requests.post", return_value=FakeExternalResponse())
+
+@pytest.fixture(scope='function')
+def fake_invalid_auth_from_oc(mocker):
+    """
+    Changes the return value of requests.post to be a custom response
+    object so that we aren't validating external APIs in our unit tests
+    """
+    class ExternalApiRequest(object):
+        @classmethod
+        def post(cls):
+            return FakeExternalResponse()
+
+    class FakeExternalResponse(object):
+        @classmethod
+        def json(self):
+            # Mock an error returned from the OC backend
+            return {'error': 'Invalid Email or password.'}
+
+    mocker.patch("requests.post", return_value=FakeExternalResponse())
+
+@pytest.fixture(scope='function')
+def fake_items_error(mocker):
+    """
+    Intentionally raises an error to test error response is correct
+    """
+    def items():
+        raise Exception("An \"unexpected\" Exception was raised!")
+
+    mocker.patch('app.utils.Paginator.items', return_value=items, side_effect=items)
+
+@pytest.fixture(scope='function')
+def fake_commit_error(mocker):
+    """
+    Intentionally raises an error to test error response is correct
+    """
+    def commit():
+        raise Exception("An \"unexpected\" Exception was raised!")
+
+    mocker.patch('app.db.session.commit', return_value=commit, side_effect=commit)
+
+@pytest.fixture(scope='function')
+def fake_key_query_error(mocker):
+    """
+    Intentionally raises an error to test error response is correct
+    """
+    def key_query():
+        raise Exception()
+
+    mocker.patch('app.models.Key.query', return_value=key_query, side_effect=key_query)
```

Comment:
Do you know if the solution was the rvalue or the side_effect? I saw things like this about using a side effect to get the mocker to see the exception.

Reply:
I don't know. I know that return_value by itself didn't seem to work, but I changed so many things tinkering with it that I'm not sure what actually pushed it over the edge. I can try taking away return_value and seeing if it still works

Reply:
Removing return_value worked, so I'll commit that. I guess all it needed was side_effect.